src/components/Form/FormWrapper.tsx

```diff
@@ -0,0 +1,175 @@
+import React, {useCallback, useMemo, useRef} from 'react';
+import type {RefObject} from 'react';
+import type {StyleProp, View, ViewStyle} from 'react-native';
+import {Keyboard, ScrollView} from 'react-native';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import FormAlertWithSubmitButton from '@components/FormAlertWithSubmitButton';
+import FormSubmit from '@components/FormSubmit';
+import SafeAreaConsumer from '@components/SafeAreaConsumer';
+import type {SafeAreaChildrenProps} from '@components/SafeAreaConsumer/types';
+import ScrollViewWithContext from '@components/ScrollViewWithContext';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import type {Form} from '@src/types/onyx';
+import type {Errors} from '@src/types/onyx/OnyxCommon';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type {FormProps, InputRefs} from './types';
+
+type FormWrapperOnyxProps = {
+    /** Contains the form state that must be accessed outside the component */
+    formState: OnyxEntry<Form>;
+};
+
+type FormWrapperProps = ChildrenProps &
+    FormWrapperOnyxProps &
+    FormProps & {
+        /** Submit button styles */
+        submitButtonStyles?: StyleProp<ViewStyle>;
+
+        /** Server side errors keyed by microtime */
+        errors: Errors;
+
+        /** Assuming refs are React refs */
+        inputRefs: RefObject<InputRefs>;
+    };
+
+function FormWrapper({
+    onSubmit,
+    children,
+    formState,
+    errors,
+    inputRefs,
+    submitButtonText,
+    footerContent,
+    isSubmitButtonVisible = true,
+    style,
+    submitButtonStyles,
+    enabledWhenOffline,
+    isSubmitActionDangerous = false,
+    formID,
+    scrollContextEnabled = false,
+    shouldHideFixErrorsAlert = false,
+}: FormWrapperProps) {
+    const styles = useThemeStyles();
+    const formRef = useRef<ScrollView>(null);
+    const formContentRef = useRef<View>(null);
+    const errorMessage = useMemo(() => (formState ? ErrorUtils.getLatestErrorMessage(formState) : undefined), [formState]);
```

Comment:
`formState` is of type `OnyxEntry<Form>` -> `Form | null`. Because formState is provided with withOnyx and all onyx values can be initially null.

> Also, why are the default props missing for some fields?
Not needed anymore.